pkg/asset/installconfig/gcp/validation.go

```diff
@@ -284,3 +285,17 @@ func validateRegion(client API, ic *types.InstallConfig, fieldPath *field.Path)
 	}
 	return nil
 }
+
+// validateCredentialMode checks whether the credential mode is
+// compatible with the authentication mode.
+func validateCredentialMode(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	creds := client.GetCredentials()
+
+	if creds.JSON == nil && ic.CredentialsMode != types.ManualCredentialsMode {
```

Comment:
Should we check the credentials mode first? If the creds mode is not correct error immediately without calling `GetCredentials()`.

Reply:
The "correctness" of the credentials depends on their contents, so we have to grab them.